sql-plugin/src/main/scala/com/nvidia/spark/rapids/HostColumnarToGpu.scala

```diff
@@ -279,6 +279,11 @@ class HostToGpuCoalesceIterator(iter: Iterator[ColumnarBatch],
     // schema and desired batch size
     batchRowLimit = GpuBatchUtils.estimateRowCount(goal.targetSizeBytes,
       GpuBatchUtils.estimateGpuMemory(schema, 512), 512)
+    // when there aren't any columns, it generally means user is doing a count() and we don't
```

Comment:
nice find.

Let us not call `estimateRowCount/estimateGpuMemory` when there no columns:
```scala
batchRowLimit = if (batch.numCols() > 0 ) {
  GpuBatchUtils.estimateRowCount(goal.targetSizeBytes,
      GpuBatchUtils.estimateGpuMemory(schema, 512), 512)
} else {
  Integer.MAX_VALUE
}      
```